dcpy/models/product/dataset/metadata.py

```diff
@@ -142,16 +147,51 @@ def get_columns(self, metadata: Metadata) -> list[Column]:
         return cols
 
 
+class ZipFile(File, extra="forbid"):
+    type: Literal["Zip"] = "Zip"
+    contains: list[File]
+
+
 class Package(BaseModel, extra="forbid"):
+    # TODO: should these just be combined into one flat list of files? Why have three distinctions
     dataset_files: list[DatasetFile]
-    attachments: list[str]
+    attachments: list[File]
+    zip_files: list[ZipFile] = []
+
+    _ASSET_TYPES = Literal["dataset_files", "attachments", "zip_files"]
+
+    def __init__(self, **data):
+        file_attachments = []
+        for a in data["attachments"]:
```

Comment:
previously we had attachments as a list of strings, which was a real pain to deal with. This warns on import, and ports them to the new format.